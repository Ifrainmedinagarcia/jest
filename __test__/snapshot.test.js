import { getCharacter } from '../snapshot'
import Rick from '../rick.json'
import { expect } from '@jest/globals';

describe('it is snapshot time', () => {
    test('should snapshot', () => {
        expect(getCharacter(Rick)).toMatchSnapshot()
    });

    test('always failed the snapshot', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20)
        }
        expect(user).toMatchSnapshot()
    });

    test('we have a exception ', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Ifrain Medina Garc√≠a'
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    });
});
