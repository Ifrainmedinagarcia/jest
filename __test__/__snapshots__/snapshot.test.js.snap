// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it is snapshot time always failed the snapshot 1`] = `
Object {
  "createAt": 2021-06-20T19:57:32.296Z,
  "id": 4,
}
`;

exports[`it is snapshot time should snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`it is snapshot time we have a exception  1`] = `
Object {
  "id": Any<Number>,
  "name": "Ifrain Medina Garc√≠a",
}
`;
